firewood/src/merkle/proof.rs

```diff
@@ -146,14 +146,20 @@ impl<N: AsRef<[u8]> + Send> Proof<N> {
         self.0.extend(other.0)
     }
 
-    pub fn verify_range_proof<K: AsRef<[u8]>, V: AsRef<[u8]>>(
+    pub fn verify_range_proof<K, V, T>(
         &self,
         root_hash: HashKey,
         first_key: K,
         last_key: K,
         keys: Vec<K>,
         vals: Vec<V>,
-    ) -> Result<bool, ProofError> {
+    ) -> Result<bool, ProofError>
+    where
+        K: AsRef<[u8]>,
+        V: AsRef<[u8]>,
+        T: BinarySerde,
+        EncodedNode<T>: serde::Serialize + serde::de::DeserializeOwned,
```

Comment:
Are there any constraints here that can be removed? Or all they all needed? Am I just missing something with where `T` gets used?

Reply:
I think we need this for our call to `InMemoryMerkle::new` which needs it for the call to `Merkle::new`